### DISCUSSION TITLE
How to use useKeyPress? It only fires every other press?

### DISCUSSION THREAD

**Discussion by raarts** [xyflow/xyflow#2010] [Q&A] [answered]
I don't understand how to use keypress. I want to make Meta-S save the graph, but it only saves the graph every other keypress, and (even worse) it fires on program start as well, saving an empty graph.

Is this wrong: 

```
  const cmdAndSPressed = useKeyPress(['Meta+s', 'Strg+s']);

  useEffect(() => {
    doSave().then();
  }, [cmdAndSPressed]);
```

**Comment by moklick** [comment, marked as answer]
We fixed some bugs for the `useKeyPress` hook. Please try the latest version. When you mount a component that uses `useKeyPress`, the returned value will be `false`. In your case you would need to test if `cmdAndSPressed` is true.

``` js
useEffect(() => {
  if (cmdAndSPressed) {
    doSave().then();
  }
}, [cmdAndSPressed]);
```
